fava 1.14 (new formula)

jacobgreenleaf opened the pull request:
This adds a formula to install fava 1.14, a web server frontend for
beancount. fava is packaged on PyPI under the same name.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

For testing, similar to #52489, we use `bean-example` to generate an example ledger. Fava is a web server, so our test starts it and sends a request using `curl`, similar to [`traefik`](https://github.com/Homebrew/homebrew-core/blob/4fb50f7484249b80795f7e57d6794dc57426d2ed/Formula/traefik%401.rb). We find an open port using `socket.TCPServer`, fork and start fava, wait 1 second for Fava to start up, and send a request to `/example-beancount-file/income_statement/` on the (hopefully) running HTTP server. Finally, we teardown the fava process.

SMillerDev commented on Formula/fava.rb:
```diff
@@ -0,0 +1,284 @@
+  sha256 "1eb7638252ba110840a94fca5a88fbd3c35057efb0146dce1258a97305be3ea0"
+  head "https://github.com/beancount/fava.git"
+
+  depends_on "python"
```
Can you try `python@3.8`

SMillerDev commented:
```
/usr/bin/sandbox-exec -f /private/tmp/homebrew20200404-57962-1abb9a7.sb ruby -W0 -I $LOAD_PATH -- /usr/local/Homebrew/Library/Homebrew/test.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/fava.rb --verbose
==> /usr/local/Cellar/fava/1.14/bin/bean-example
Error: fava: failed
An exception occurred within a child process:
  Errno::ENOENT: No such file or directory - /usr/local/Cellar/fava/1.14/bin/bean-example
```

jacobgreenleaf commented:
Thanks, I accidentally had a test dependency on `beancount` to use `bean-example`. I have removed it and used an inline checked-in fixture for the test ledger.

SMillerDev commented on Formula/fava.rb:
```diff
@@ -0,0 +1,290 @@
+    require "socket"
+
+    # Find an available port
+    http_server = TCPServer.new(0)
+    http_port = http_server.addr[1]
+    http_server.close
```
see https://github.com/Homebrew/homebrew-core/pull/52595
```suggestion
    http_port = free_port
```

jacobgreenleaf commented:
Updated to use `free_port`. 

I am guessing the CI is failing due to `sleep 1`. In line with the existing tests I've just adjusted the sleep time to `5`, but it would be useful if we had a few helpers like `wait_for_tcp_port`.

jacobgreenleaf commented:
Fixed formatting issues with the heredoc statement.

SMillerDev commented:
@Bo98 is this due to recent changes?
```
    ERROR: b'xcrun: error: missing DEVELOPER_DIR path: \n'
    ** make sure the development packages of libxml2 and libxslt are installed **
```

Bo98 commented:
> DEVELOPER_DIR path: \n

what on earth

Bo98 commented:
@BrewTestBot test this please

jacobgreenleaf commented:
@Bo98 @SMillerDev any update?
